packages/web-components/src/avatar/avatar.base.ts

```diff
@@ -37,4 +43,20 @@ export class BaseAvatar extends FASTElement {
 
     this.elementInternals.role = 'img';
   }
+
+  /**
+   * Removes any empty text nodes from the default slot when the slotted content changes.
+   *
+   * @param e - The event object
+   * @internal
+   */
+  public slotchangeHandler(e: Event): void {
+    if (!this.innerText.trim()) {
+      this.defaultSlot.assignedNodes().forEach(node => {
+        if (node.nodeType === Node.TEXT_NODE) {
+          (node as Element).remove();
```

Comment:
Does editing the slot inside the slotchange event cause the slotchange event to fire again?